scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Check for decrease in coverage from 100% of frontend files."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+import python_utils
+
+from . import fully_covered_files
+
+LCOV_FILE_PATH = os.path.join(os.pardir, 'karma_coverage_reports', 'lcov.info')
+PR_LCOV_FILE_PATH = os.path.join(os.curdir, 'tmp', 'pr-lcov.info')
+FULLY_COVERED_TESTS = fully_covered_files.fully_covered_files_array
+
+
+def run_frontend_tests_script():
+    """Run the frontend tests script using subprocess."""
+    subprocess.check_call([
+        'python', '-m', 'scripts.run_frontend_tests'])
+
+
+def create_tmp_folder():
+    """Creates a temporary folder."""
+    if not os.path.exists('./tmp'):
+        os.mkdir('./tmp')
+
+
+def delete_tmp_folder():
+    """Delete the temporary folder."""
+    if os.path.exists('./tmp'):
+        shutil.rmtree('./tmp')
+
+
+def filter_lines(line):
+    """Check if the line has file path (SF) or total lines (LF) or covered
+    lines (LH) of the test.
+
+    Args:
+        line: str. A line from lcov file.
+
+    Returns:
+        Boolean. If the line has the file path or total lines or covered lines
+        of the test.
+    """
+    return ('SF' in line or 'LH' in line or 'LF' in line)
+
+
+def get_test_file_name(test_path):
+    """Get the file name from the absolute path.
+
+    Args:
+        test_path: string. The file's absolute path.
+
+    Returns:
+        String. The file path.
+    """
+    if not test_path:
+        sys.stderr.write(
+            'The test path is empty or null.'
+            'It\'s not possible to diff the test coverage correctly.')
+        sys.exit(1)
+
+    file_name = os.path.split(test_path)[1]
+    return file_name
+
+
+def get_lcov_file_tests(file_path):
+    """Get all tests from a lcov file, and filters it to return only:
+    - File path
+    - File total lines
+    - File covered lines
+
+    Args:
+        file_path: string. The path of lcov file.
+
+    Returns:
+        String array. An array with all tests filtered, including only important
```

Comment:
Typeinfo should be list(str). Please be careful to follow existing codebase conventions.

Reply:
Sorry I'm still little confused by Python variable types.

Reply:
Done.